pandas/_config/config.py

```diff
@@ -765,14 +770,14 @@ def is_type_factory(_type):
 
     """
 
-    def inner(x):
+    def inner(x) -> None:
         if type(x) != _type:
             raise ValueError(f"Value must have type '{_type}'")
 
     return inner
 
 
-def is_instance_factory(_type):
+def is_instance_factory(_type) -> Callable[..., None]:
```

Comment:
the returned function accepts exactly one argument

```suggestion
def is_instance_factory(_type) -> Callable[[Any], None]:
```